### PR TITLE
Added code comments

### PR DESCRIPTION
**Pull request by savan-chovatiya**
# Description of change
- Added code comments for better readability

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_pendo/__init__.py [hunk 1]
```diff
@@ -39,6 +42,7 @@ class DependencyException(Exception):
 
 
 def validate_dependencies(selected_stream_ids):
+    # Validate and raise exceptions if sub-streams are selected but related parents not selected
```
Comment (umangagrawal-crest) [inline tap_pendo/__init__.py:45]: Validate and raise exceptions if sub-streams are selected but related parents are not selected

File: tap_pendo/sync.py [hunk 2]
```diff
@@ -38,18 +41,21 @@ def sync_stream(state, start_date, instance):
                 LOGGER.error('Transform failed for %s', record)
                 raise err
 
+            # Get replication key's value from record and update bookmark with max
```
Comment (umangagrawal-crest) [inline tap_pendo/sync.py:44]: Get the value of replication key from record and update bookmark with max

File: tap_pendo/streams.py [hunk 3]
```diff
@@ -399,13 +412,15 @@ def sync_substream(self, state, parent, sub_stream, parent_response):
...
             try:
                 with metrics.record_counter(
                         sub_stream.name) as counter, Transformer(
                             integer_datetime_fmt=
                             "unix-milliseconds-integer-datetime-parsing"
                         ) as transformer:
+                    # Get records of sub-stream
                     stream_events = sub_stream.sync(state, new_bookmark,
                                                     record.get(parent.key_properties[0]))
                     for event in stream_events:
```
[the remark this answers is not in the document]
Comment (savan-chovatiya) [reply tap_pendo/streams.py:428]: Added more comments above as per suggestion.

File: tap_pendo/__init__.py [hunk 4]
```diff
@@ -17,17 +17,20 @@
 
 
 def do_discover(config):
+    # Discover schemas for all streams and dump catalog
```
[the remark this answers is not in the document]
Comment (karanpanchal-crest) [reply tap_pendo/__init__.py]: Updated the comment

**Pull request closed without merging**